recipes/libssh2/all/conanfile.py

```diff
@@ -51,31 +48,36 @@ def configure(self):
     def requirements(self):
         if self.options.with_zlib:
             self.requires.add("zlib/1.2.11")
-        if self.options.with_openssl:
-            self.requires.add("openssl/1.0.2t")
+        if self.options.crypto_backend == "openssl":
+            self.requires.add("openssl/1.1.1d")
+        elif self.options.crypto_backend == "mbedtls":
+            self.requires.add("mbedtls/2.16.3-gpl")
 
-    def _configure_cmake(self):
+    def _get_cmake(self):
         cmake = CMake(self)
         cmake.definitions['BUILD_SHARED_LIBS'] = self.options.shared
         cmake.definitions['ENABLE_ZLIB_COMPRESSION'] = self.options.with_zlib
         cmake.definitions['ENABLE_CRYPT_NONE'] = self.options.enable_crypt_none
         cmake.definitions['ENABLE_MAC_NONE'] = self.options.enable_mac_none
-        if self.options.with_openssl:
+        if self.options.crypto_backend == "openssl":
             cmake.definitions['CRYPTO_BACKEND'] = 'OpenSSL'
             cmake.definitions['OPENSSL_ROOT_DIR'] = self.deps_cpp_info['openssl'].rootpath
+        elif self.options.crypto_backend == "mbedtls":
+            cmake.definitions['CRYPTO_BACKEND'] = "mbedTLS"
         else:
             raise ConanInvalidConfiguration("Crypto backend must be specified")
+
         cmake.definitions['BUILD_EXAMPLES'] = False
         cmake.definitions['BUILD_TESTING'] = False
-        cmake.configure()
         return cmake
 
     def build(self):
-        cmake = self._configure_cmake()
+        cmake = self._get_cmake()
```

Comment:
why renamed?

Reply:
I moved the `cmake.configure()` out of the `_configure_cmake` function.
Because it did not configure anymore, I renamed.

I did this because cmake configuration should only happen 1 time. During build.
The `package` function should only run the install step.

For some packages, cmake configuration takes a long time.
And as this is unnecessary, just avoid it.

Reply:
I've renamed it back to the original name.